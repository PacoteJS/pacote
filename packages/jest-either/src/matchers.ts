export interface AsymmetricMatcher {
  asymmetricMatch(other: any): boolean
}

export function isAsymmetricMatcher(
  matcher: any
): matcher is AsymmetricMatcher {
  return typeof matcher.asymmetricMatch === 'function'
}
