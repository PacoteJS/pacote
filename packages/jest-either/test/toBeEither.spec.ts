import { left, right } from 'fp-ts/lib/Either'
import matchers from '../src/index'

expect.extend(matchers)

describe('.toBeEither()', () => {
  test('passes when object is a left', () => {
    expect(left(true)).toBeEither()
  })

  test('passes when object is a right', () => {
    expect(right(true)).toBeEither()
  })

  test('fails when object is undefined', () => {
    expect(() => expect(undefined).toBeEither()).toThrow()
  })

  test('fails when object is not an Either', () => {
    expect(() => expect(true).toBeEither()).toThrow()
  })
})

describe('.not.toBeEither()', () => {
  test('passes when object is not an Either', () => {
    expect(true).not.toBeEither()
  })

  test('passes when object is undefined', () => {
    expect(undefined).not.toBeEither()
  })

  test('fails when object is a left', () => {
    expect(() => expect(left(false)).not.toBeEither()).toThrow()
  })

  test('fails when object is a right', () => {
    expect(() => expect(right(false)).not.toBeEither()).toThrow()
  })
})
