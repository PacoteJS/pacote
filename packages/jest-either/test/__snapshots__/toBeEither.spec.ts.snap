// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeEither() fails when object is a left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toBeEither()[22m

Unexpected Either, received:
  [31m{\\"_tag\\": \\"Left\\", \\"value\\": false}[39m"
`;

exports[`.not.toBeEither() fails when object is a right 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toBeEither()[22m

Unexpected Either, received:
  [31m{\\"_tag\\": \\"Right\\", \\"value\\": false}[39m"
`;

exports[`.toBeEither() fails when object is not an Either 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeEither()[22m

Expected Either, received:
  [31mtrue[39m"
`;

exports[`.toBeEither() fails when object is undefined 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeEither()[22m

Expected Either, received:
  [31mundefined[39m"
`;
